receiver/otlpreceiver/otlp_test.go

```diff
@@ -393,15 +393,19 @@ func TestStartWithoutConsumersShouldFail(t *testing.T) {
 }
 
 func tempSocketName(t *testing.T) string {
-	tmpfile, err := ioutil.TempFile("", "sock")
-	require.NoError(t, err)
-	socket := tmpfile.Name()
-	err = os.Remove(socket)
+	socket := tempFile(t)
+	err := os.Remove(socket)
 	require.NoError(t, err)
-
 	return socket
 }
 
+func tempFile(t *testing.T) string {
+	tmpfile, err := ioutil.TempFile("", "sock")
+	require.NoError(t, err)
+	defer tmpfile.Close()
```

Comment:
Why not calling close directly?

```
require.NoError(t, tmpfile.Close())
```

Reply:
I guess I thought it was idiomatic in Go to always `defer x.Close()` when using an `io.Closer` (I didn't actually consider that this ignores the errors so may not always be a good idea, e.g. in tests 😮). Anyway, changed as per your suggestion